coaddAnalysis.py

```diff
@@ -0,0 +1,234 @@
+
+    xMin = min(max(tractRa), patchBoundary.raMax) + buff
+    xMax = max(min(tractRa), patchBoundary.raMin) - buff
+    yMin = max(min(tractDec), patchBoundary.decMin) - buff
+    yMax = min(max(tractDec), patchBoundary.decMax) + buff
+    xlim = xMin, xMax
+    ylim = yMin, yMax
+    axes.fill(tractRa, tractDec, fill=True, edgecolor='k', lw=1, linestyle='solid',
+              color="black", alpha=0.2)
+    for ip, patch in enumerate(tractInfo):
+        patchIndexStr = str(patch.getIndex()[0]) + "," + str(patch.getIndex()[1])
+        color = "k"
+        alpha = 0.05
+        if patchIndexStr in patchList:
+            # color = ("c", "g", "r", "b", "m")[ip%5]
+            color = 'g'
+            alpha = 0.5
+        ra, dec = bboxToRaDec(patch.getOuterBBox(), tractInfo.getWcs())
+        deltaRa = abs(max(ra) - min(ra))
+        deltaDec = abs(max(dec) - min(dec))
+        pBuff = 0.5*max(deltaRa, deltaDec)
+        centerRa = min(ra) + 0.5*deltaRa
+        centerDec = min(dec) + 0.5*deltaDec
+        if (centerRa < xMin + pBuff and centerRa > xMax - pBuff and
+                centerDec > yMin - pBuff and centerDec < yMax + pBuff):
+            axes.fill(ra, dec, fill=True, color=color, lw=1, linestyle="solid", alpha=alpha)
+            if patchIndexStr in patchList or (centerRa < xMin - 0.2*pBuff and
+                                              centerRa > xMax + 0.2*pBuff and
+                                              centerDec > yMin + 0.2*pBuff and
+                                              centerDec < yMax - 0.2*pBuff):
+                axes.text(percent(ra), percent(dec, 0.5), str(patchIndexStr),
+                          fontsize=fontSize - 1, horizontalalignment="center", verticalalignment="center")
+    axes.set_xlabel('RA', size='large')
+    axes.set_ylabel('Dec', size='large')
+    axes.set_xlim(xlim)
+    axes.set_ylim(ylim)
+
+
+def mosaicCoadd(butler, patch_list, band='g', tract=0, ref_patch=None, sampling=100, norm=None, nImage=False,
+                fig=None, show_colorbar=True, filename=None):
+    """Generate a mosaic image of many coadded patches.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.base.Butler`
+        A Butler instance to retrieve data from a repository.
+    patch_list : `list` of `str`
+        A list of the patches containing images to mosaic.
+    band : `str`, optional
+        The band of the coadd to retrieve from the repository.
+    tract : `int`, optional
+        The tract of the skyMap.
+    ref_patch : `str`, optional
+        If set, use the given patch to set the image normalization for the figure.
+    sampling : `int`, optional
+        Stride factor to sample each input image in order to reduce the size in memory.
+        A `sampling` of 1 will attempt to display every pixel.
+    norm : `astropy.visualization.ImageNormalize`, optional
+        Normalization to set the color scale of the images.
```

Comment:
I think this would be clearer if the default were called `zscale` and not `None`

Reply:
That's not quite what is going on here. This isn't choosing the type of normalization, it is a full normalization object. So, it also contains the parameters of the scaling and not just the type. I will add to the description to make this more clear.